### ISSUE TITLE
bot doesn't accept friend requests sent when bot is offline

### ISSUE THREAD

**Issue by MajorErection** [steam-forward/node-steam-forum#49]
My bot doesn't accept any friend requests that are sent when my bot is offline, I think I'm missing something but I can't figure it out, this is my code

     steamFriends.on('friend', function(steamID, relationship) {
             if (relationship == Steam.EFriendRelationship.RequestRecipient) {
                 console.log ('[' + steamID + '] Accepted friend request');
                 steamFriends.addFriend(steamID);
             }
             else if (relationship == Steam.EFriendRelationship.None) {
                 console.log('[' + steamID + '] Un-friended');
             }
         });

**Comment by scholtzm** [comment]
See [relationships](https://github.com/seishun/node-steam/tree/master/lib/handlers/friends#relationships) event.

**Comment by MajorErection** [comment]
I'm pretty sure this is wrong but at least its an attempt I guess :/ 

         steamFriends.on('relationships', function (steamID, friend) {
                              
                 if (friend == Steam.EFriendRelationship.RequestRecipient) {
                     console.log("Friend request while offline from: " + steamID);
                     steamFirends.addFriend(steamID);
                 }
         });

**Comment by scholtzm** [comment]
Close, but no cigar.

The callback has no arguments, you have to traverse `steamFriends.friends` property and check status for each friend.

https://github.com/seishun/node-steam/tree/master/lib/handlers/friends#friends

**Comment by MajorErection** [comment]
sorry but I don't quite understand what you mean, if you don't mind can you rephrase in a simpler explanation for me?

**Comment by scholtzm** [comment]
See the code here: https://github.com/scholtzm/vapor/blob/master/lib/handlers/relationships.js

**Comment by dragonbanshee** [comment]
Old but I figure I'd post it here to help:

```javascript
steamFriends.on('relationships', function() {
    for(var friend in steamFriends.friends) {
        if(steamFriends.friends[friend] === Steam.EFriendRelationship.RequestRecipient) {
            steamFriends.addFriend(friend);
        }
    }
});
```

Change ` steamFriends.addFriend(friend); ` to whatever you want to do if you get a friend request while offline.
